bulk_indexer.go

```diff
@@ -389,7 +390,27 @@ func (b *BulkIndexer) Flush(ctx context.Context) (BulkIndexerResponseStat, error
 	b.bytesUncompFlushed = bytesUncompFlushed
 	var resp BulkIndexerResponseStat
 	if res.IsError() {
-		e := errorFlushFailed{resp: res.String(), statusCode: res.StatusCode}
+		var s string
+		var er struct {
+			Error struct {
+				Type     string `json:"type,omitempty"`
+				Reason   string `json:"reason,omitempty"`
+				CausedBy struct {
+					Type   string `json:"type,omitempty"`
+					Reason string `json:"reason,omitempty"`
+				} `json:"caused_by,omitempty"`
+			} `json:"error,omitempty"`
+		}
+
+		if err := jsoniter.NewDecoder(res.Body).Decode(&er); err == nil {
```

Comment:
if an error was returned here when decoding, should we consider fallback to `s = res.String()`?